core/src/main/java/org/apache/iceberg/rest/RESTContentFileParser.java

```
@@ -0,0 +1,250 @@
+    Preconditions.checkArgument(
+        jsonNode.isObject(), "Invalid JSON node for content file: non-object (%s)", jsonNode);
+
+    int specId = JsonUtil.getInt(SPEC_ID, jsonNode);
+    FileContent fileContent = FileContent.valueOf(JsonUtil.getString(CONTENT, jsonNode));
+    String filePath = JsonUtil.getString(FILE_PATH, jsonNode);
+    FileFormat fileFormat = FileFormat.fromString(JsonUtil.getString(FILE_FORMAT, jsonNode));
+
+    // TODO at the time of deserialization we dont have the partition spec we just have spec id.
+    // we will need to get the spec from table metadata using spec id.
+    // we wil need to send parition spec i believe, put some placeholder here for now null
+    PartitionData partitionData = new PartitionData();
+
+    long fileSizeInBytes = JsonUtil.getLong(FILE_SIZE_IN_BYTES, jsonNode);
+    Metrics metrics = metricsFromJson(jsonNode);
+    ByteBuffer keyMetadata = JsonUtil.getByteBufferOrNull(KEY_METADATA, jsonNode);
+    List<Long> splitOffsets = JsonUtil.getLongListOrNull(SPLIT_OFFSETS, jsonNode);
+    int[] equalityFieldIds = JsonUtil.getIntArrayOrNull(EQUALITY_IDS, jsonNode);
+    Integer sortOrderId = JsonUtil.getIntOrNull(SORT_ORDER_ID, jsonNode);
+
+    if (fileContent == FileContent.DATA) {
+      return new GenericDataFile(
+          specId,
+          filePath,
+          fileFormat,
+          partitionData,
+          fileSizeInBytes,
+          metrics,
+          keyMetadata,
+          splitOffsets,
+          sortOrderId);
+    } else {
+      return new GenericDeleteFile(
+          specId,
+          fileContent,
+          filePath,
+          fileFormat,
+          partitionData,
+          fileSizeInBytes,
+          metrics,
+          equalityFieldIds,
+          sortOrderId,
+          splitOffsets,
+          keyMetadata);
+    }
+  }
+
+  private static void metricsToJson(ContentFile<?> contentFile, JsonGenerator generator)
```

Q: will we contain stats for the columns being projected or for all columns of data files ? coming mostly with an intention to reduce payload

---

I think we have this `statsFields` which is in the request of the https://github.com/apache/iceberg/blob/main/open-api/rest-catalog-open-api.yaml#L4231 of the `planTableScan` operation, and specifies which columns to send stats for.

---

Got you, some follow-ups, This implies statsFields will be a subset of projected columns?
1/ whats the default, all projected columns or all columns of table ?
2/ what should be response if we requests stats for columns not being projected ?

---

@singhpk234 Thanks for bringing up these points!

1. Currently based on the spec we don't specify a default. My understanding is that the `stats-fields` would be bounded by what was passed in the `select` here https://github.com/apache/iceberg/blob/main/open-api/rest-catalog-open-api.yaml#L4198 

2. I think in this case we should probably throw an exception (if my above understanding is correct). Potentially we may need to introduce a new exception for this case?

cc @amogh-jahagirdar